File: 문자열/5430 - AC.py
import sys
read = sys.stdin.readline

t = int(read())

result = []

for i in range(t) :
    p = read().strip()
    n = int(read())
    arr = read().strip()
    if arr != '[]' :
        arr = list(map(int, arr[1:-1].split(',')))
    else :
        arr = []

    idx = 0
    delete = [0, 0]
    head = 0
    stack = []
    while True :
        if idx == len(p) :
            break
        if p[idx] == 'R' :
            if len(stack) > 0 and stack[-1] == 'R':
                stack.pop()
                head = 0
            else :
                head = 1
                stack.append(p[idx])
        else :
            delete[head] += 1
        idx += 1
    
    if sum(delete) > n :
        result.append('error')
        continue
    if delete[1] == 0 :
        arr = arr[delete[0]:]
    else :
        arr = arr[delete[0]:-delete[1]]
    if len(stack) != 0 :
        arr.reverse()
    string = '['
    string += ','.join(map(str, arr))
    string += ']'
    result.append(string)

print('\n'.join(map(str, result)))

File: 문자열/5430 - AC.js
const fs = require('fs');
const input = fs
  .readFileSync('./문자열/5430input.txt')
  .toString()
  .trim()
  .split('\r\n');
// const input = fs.readFileSync('dev/stdin').toString().trim().split('\n');

const t = +input[0];

const arrayToString = (arr) => {
  let s = '[';
  s += arr.join(',');
  s += ']';
  return s;
};

const slice = (r, left, right, arr) => {
  if (left - right >= 2) {
    return 'error';
  }
  const newArr = arr.slice(left, right + 1);
  if (!r) {
    return arrayToString(newArr);
  }
  return arrayToString(newArr.reverse());
};

const ac = (p, n, arr) => {
  let r = false;
  let left = 0;
  let right = n - 1;
  for (let i = 0; i < p.length; i++) {
    if (p[i] === 'R') {
      r = !r;
      continue;
    }
    if (!r && p[i] === 'D') {
      left++;
      continue;
    }
    right--;
  }
  console.log(slice(r, left, right, arr));
};

for (let i = 0; i < t; i++) {
  const p = input[i * 3 + 1];
  const n = +input[i * 3 + 2];
  const arr = input[i * 3 + 3].slice(1, -1).split(',').map(Number);
  ac(p, n, arr);
}
